packages/relay/src/RelayServer.ts

```diff
@@ -780,31 +786,14 @@ latestBlock timestamp   | ${latestBlock.timestamp}
     }
   }
 
-  _getLatestTxBlockNumber (): number {
-    return this.lastMinedActiveTransaction?.blockNumber ?? -1
-  }
-
-  _getLatestRegisterTxBlockTimestamp (): number {
-    return this.registrationManager.lastMinedRegisterTransaction?.blockTimestamp ?? -1
-  }
-
-  async _updateLatestTxBlockNumber (eventsSinceLastScan: EventData[]): Promise<void> {
-    const latestTransactionSinceLastScan = getLatestEventData(eventsSinceLastScan)
-    if (latestTransactionSinceLastScan != null) {
-      this.lastMinedActiveTransaction = latestTransactionSinceLastScan
-      this.logger.debug(`found newer block ${this.lastMinedActiveTransaction?.blockNumber}`)
-    }
-    if (this.lastMinedActiveTransaction == null) {
-      this.lastMinedActiveTransaction = await this._queryLatestActiveEvent()
-      this.logger.debug(`queried node for last active server event, found in block ${this.lastMinedActiveTransaction?.blockNumber}`)
-    }
-  }
-
   async _queryLatestActiveEvent (): Promise<EventData | undefined> {
-    const events: EventData[] = await this.contractInteractor.getPastEventsForHub([address2topic(this.managerAddress)], {
-      fromBlock: this.config.coldRestartLogsFromBlock
+    const hubEvents: EventData[] = await this.contractInteractor.getPastEventsForHub([address2topic(this.managerAddress)], {
+      fromBlock: this.lastScannedBlock + 1
+    })
+    const registrarEvents: EventData[] = await this.contractInteractor.getPastEventsForRegistrar([address2topic(this.managerAddress)], {
```

Comment:
why would you care about "registrar" events?

Reply:
Because that was a bug, since it includes registration events, which registrationManager in the server still uses.